hooks/pre_gen_project.py

```diff
@@ -1,23 +1,46 @@
 import re
 import sys
+from pathlib import Path
 
-PROJECT_REGEX = r'^[-a-zA-Z][-a-zA-Z0-9]+$'
-project_slug = '{{ cookiecutter.project_slug }}'
 
-if not re.match(PROJECT_REGEX, project_slug):
-    print('ERROR: %s is not a valid project slug. Slug can only contain letters, digits, '
-          'and dashes.' % project_slug)
+PROJECT_DIR = "{{ cookiecutter.project_dir }}"
+
+FULL_PROJECT_DIR = (Path(".") / PROJECT_DIR).resolve()
+FORBIDDEN_CHARS = r'<>:"\/|?*'
+
+forbidden_chars_in_dir_name = any(ch in PROJECT_DIR for ch in FORBIDDEN_CHARS)
+
+if forbidden_chars_in_dir_name:
+    print(f"ERROR: '{PROJECT_DIR}' contains forbidden chars ({FORBIDDEN_CHARS})")
     sys.exit(1)
+elif not FULL_PROJECT_DIR.exists():
+    try:
+        FULL_PROJECT_DIR.mkdir(parents=True)
+    except Exception:
+        print(
+            f"ERROR: '{FULL_PROJECT_DIR}' is not a valid project directory name "
+            "since OS cannot create it."
+        )
+        sys.exit(1)
+    else:
+        FULL_PROJECT_DIR.rmdir()
```

Comment:
not sure i understand your intention here. could you please explain

Reply:
I was thinking that cookiecutter will raise an exception that the folder existrs. However I've tested that - all is ok, removed that else block in c7d7b05